enterprise/utils.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+Utilities to be used across the enterprise djangoapp
+"""
+try:
+    from openedx.core.djangoapps.api_admin.utils import course_discovery_api_client
+except ImportError:
+    course_discovery_api_client = None
+
+class NotConnectedToOpenEdX(Exception):
+    """
+    Exception that is raised when we try to import from
+    the core openedx repository, but it isn't available
+    """
+    pass
+
+
+def get_catalog_api_client(user):
+    """
+    Retrieve an authenticated API client that can be used to access the course
+    discovery API; raise an exception to be caught at a higher level if the
+    package doesn't have OpenEdX resources available.
+    """
+    if course_discovery_api_client is None:
+        raise NotConnectedToOpenEdX(
+            'To get a catalog API client, this package must be installed in an OpenEdX environment.'
+        )
+    else:
+        return course_discovery_api_client(user)
+
+
+def get_all_catalogs(user):
+    """
+    Return a list of all course catalogs, including name and ID
+    """
+    # Initialize the permanent list to hold all results, as well as
+    # the ephemeral list to store each API call's results.
+    results = []
+    this_result = []
+    # Set the initial offset and limit used in API calls
+    offset = 0
+    limit = 50
+    # Initialize an API client
+    client = get_catalog_api_client(user)
+    # Loop while either we're at the beginning or we're still getting results
+    while offset == 0 or len(this_result):
```

Comment:
If we are ultimately using client `EdxRestApiClient` and our goal is to get all paginated results then this is already handled here: https://github.com/edx/edx-platform/blob/master/openedx/core/lib/edx_api_utils.py#L92
So you can use method `get_edx_api_data` to get complete data from all pages. Example using can be found here: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/catalog/utils.py#L50-L58